### PR TITLE
add validate address rpc call + tests

### PR DESCRIPTION
**Pull request by ixje**
**What current issue(s) does this address?, or what feature is it adding?**
add validate address rpc call + tests

address part of #189 

**How did you solve this problem?**
implement call + tests

**How did you make sure your solution works?**
tests

**Did you add any tests?**
yes

**Are there any special changes in the code that we should be aware of?**
no

**Did you run `make lint` and `make test`?**
yes, no

**Are you making a PR to a feature branch or development rather than master?**
feature

### CODE REVIEW

File: neo/api/JSONRPC/JsonRpcApi.py [hunk 1]
```diff
@@ -128,6 +129,21 @@ def parse_uint_str(self, param):
             return param[2:]
         return param
 
+    def validateaddress(self, params):
+        # check for [] parameter or [""]
+        if params is None or params[0] == '':
```
Comment (metachris) [inline neo/api/JSONRPC/JsonRpcApi.py]: this code doesn't yet check for an empty list (it is not None). perhaps upgrade to `if params is None or len(params) == 0 or params[0] == '':`

File: neo/api/JSONRPC/JsonRpcApi.py [hunk 2]
```diff
@@ -237,8 +253,7 @@ def json_rpc_method_handler(self, method, params):
             raise NotImplementedError()
 
         elif method == "validateaddress":
-            raise NotImplementedError()
-
+            return self.validateaddress(params)
```
Comment (metachris) [inline neo/api/JSONRPC/JsonRpcApi.py]: one line whitespace before the next elif would be nice

**Comment by metachris** [comment]
very nice 👍 
just 2 minor comments

File: neo/api/JSONRPC/JsonRpcApi.py (hunk 1 above)
Comment (ixje) [reply neo/api/JSONRPC/JsonRpcApi.py]: I should have anticipated this as I initially was surprised as well.  However, this code is correct given the current `gen_rpc_req` implementation. Here's why

```
req = self._gen_rpc_req("validateaddress", params=[])
```
returns a `dict` without the `params` keyword, therefor `if params is None` actually tests for the empty list. I'm assuming an actual request will be parsed similarly.

```
In [3]: def test_list(params=None):
   ...:     if params:
   ...:         print("yes params")
   ...:     else:
   ...:         print("no params")
   ...:

In [4]: test_list([])
no params
```

Comment (metachris) [reply neo/api/JSONRPC/JsonRpcApi.py]: I understand what you are saying. but this is not 100% correct. `if not params` matches an empty list, `if params is None` doesn't. you can just replace this if here with `if not params or params[0] == '':` for the intended effect.

**Comment by localhuman** [comment]
The way neon-js and c# checks valid address is to convert to scripthash and then back.  if it comes back the same it is valid

**Comment by ixje** [comment]
@localhuman yeah I'm aware there are much better ways to do it, but I just followed the `RpcServer.cs` code [from here](https://github.com/neo-project/neo/blob/master/neo/Network/RPC/RpcServer.cs#L240) to [here](https://github.com/neo-project/neo/blob/master/neo/Wallets/Wallet.cs#L399-L406). The parts you talk about is in the actual Wallet code (I do the same in my [mobile wallet](https://github.com/CityOfZion/neon-wallet-react-native/blob/master/app/api/crypto/index.js#L44))

**Comment by ixje** [comment]
@localhuman I'd like to keep the current validation test until we've implemented `isValidPublicAddress` discussed [here](https://github.com/CityOfZion/neo-python-core/issues/20) in `neo-python-core`. That can then be the extensive version that also does the checksum validation you mentioned. Would that work for you?

**Comment by localhuman** [comment]
That works for me.  Would you mind fixing conflicts :)

**Comment by localhuman** [comment]
best commit message ever 🤣

**Pull request merged**